lib/bridge/membershiplists.js

```diff
@@ -68,14 +69,15 @@ module.exports.getChannelsToJoin = function(server) {
                 });
             }));
         });
-        q.allSettled(promises).then(function() {
+        var onDone = function() {
             var chans = Object.keys(channels);
             log.debug(
                 "getChannelsToJoin => %s should be synced: %s",
                 chans.length, JSON.stringify(chans)
             );
             defer.resolve(chans);
-        });
+        };
+        Promise.all(promises).then(onDone, onDone);
```

Comment:
Implemented allSettled for realsies.

I can see promiseutil getting more used in the future :(